atc/engine/check_delegate.go

```diff
@@ -111,24 +111,37 @@ func (d *checkDelegate) WaitToRun(ctx context.Context, scope db.ResourceConfigSc
 		}
 	}
 
-	end, err := scope.LastCheckEndTime()
-	if err != nil {
-		if releaseErr := lock.Release(); releaseErr != nil {
-			logger.Error("failed-to-release-lock", releaseErr)
-		}
-
-		return nil, false, fmt.Errorf("get last check end time: %w", err)
-	}
-
-	runAt := end.Add(interval)
-
 	shouldRun := false
-	if d.build.IsManuallyTriggered() {
-		// ignore interval for manually triggered builds
-		shouldRun = true
-	} else if !d.clock.Now().Before(runAt) {
-		// run if we're past the last check end time
+	if d.plan.Resource == "" && d.plan.ResourceType == "" {
 		shouldRun = true
+	} else {
+		if d.build.IsManuallyTriggered() {
```

Comment:
would be nice to avoid extra indent level here:

```suggestion
	} else if d.build.IsManuallyTriggered() {
```